umbraco/Umbraco-CMS
# Fix for icon validation bug

Fix required image validation when saving in a blocklist

### Prerequisites

- [ ] I have added steps to test this contribution in the description 

If there's an existing issue for this PR then this fixes #12653

### Description

The media picker required validation was not working client side.

Required was set like this:

`<input type="hidden" name="modelValue" ng-model="vm.model.value.length" ng-required="vm.model.validation.mandatory && vm.model.value.length === 0" />`

this checks if the mandatory option is true for the property and if none has already been selected. The issue with this solution is that it does not update or check validation. I have copied a validateMandatory function from colorpicker. Now the validation works inside a block list editor.

### How to test
Make an block list item, which contains a required image field. Like in the screenshot. Leave the field empty, click create, you should see an error.

![image](https://user-images.githubusercontent.com/36473707/190871626-436fe7c2-19d7-4152-bd20-36c2d734bf35.png)
